library/ssl_tls12_server.c

```diff
@@ -931,9 +938,11 @@ static int ssl_parse_client_hello( mbedtls_ssl_context *ssl )
      * otherwise read it ourselves manually in order to support SSLv2
      * ClientHello, which doesn't use the same record layer format.
      */
+    renegotiating = 0;
```

Comment:
Why is the declaration 10 lines above rather than here?

Reply:
A variable cannot be declared directly after a label definition (something I did not know until I got the compiler error), so it has to be declared above line 934.

Reply:
A label has to be before a [statement](https://en.cppreference.com/w/c/language/statements). A declaration isn't a statement. It's a syntactic restriction which has become more annoying in C99. You can work around it by adding a semicolon after the label: that creates an empty statement, which can be jumped to.

Reply:
Ah, makes sense, thanks for the explanation.